From 10e1af7439fbd8808a820275e28c4d49ce126d1b Mon Sep 17 00:00:00 2001
Message-Id: <10e1af7439fbd8808a820275e28c4d49ce126d1b.1671091428.git.stefan@agner.ch>
In-Reply-To: <be37daf8efa7567bebbc38b5f292826324da2611.1671091428.git.stefan@agner.ch>
References: <be37daf8efa7567bebbc38b5f292826324da2611.1671091428.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Wed, 14 Dec 2022 16:36:09 +0100
Subject: [PATCH 4/6] [rest] autostart Thread network when setting active
 dataset

Start the Thread network automatically when setting the dataset
initially.
---
 src/rest/resource.cpp | 6 +++++-
 src/rest/resource.hpp | 1 +
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/rest/resource.cpp b/src/rest/resource.cpp
index 1f0b6ea540..37ceff71c2 100644
--- a/src/rest/resource.cpp
+++ b/src/rest/resource.cpp
@@ -136,7 +136,8 @@ Resource::Resource(ControllerOpenThread *aNcp)
 
 void Resource::Init(void)
 {
-    mInstance = mNcp->GetThreadHelper()->GetInstance();
+    mThreadHelper = mNcp->GetThreadHelper();
+    mInstance = mNcp->GetInstance();
 }
 
 void Resource::Handle(Request &aRequest, Response &aResponse) const
@@ -544,6 +545,9 @@ void Resource::SetActiveDatasetTlvs(const Request &aRequest, Response &aResponse
 
     errorCode = GetHttpStatus(HttpStatusCode::kStatusAccepted);
     aResponse.SetResponsCode(errorCode);
+
+    SuccessOrExit(mThreadHelper->TryResumeNetwork());
+
 exit:
     if (error != OT_ERROR_NONE)
     {
diff --git a/src/rest/resource.hpp b/src/rest/resource.hpp
index 057cafc3de..460a8926c4 100644
--- a/src/rest/resource.hpp
+++ b/src/rest/resource.hpp
@@ -139,6 +139,7 @@ private:
     void        DiagnosticResponseHandler(otError aError, const otMessage *aMessage, const otMessageInfo *aMessageInfo);
 
     otInstance           *mInstance;
+    otbr::agent::ThreadHelper   *mThreadHelper;
     ControllerOpenThread *mNcp;
 
     std::unordered_map<std::string, ResourceHandler>         mResourceMap;
-- 
2.39.0

